ns1/data_source_record.go

```diff
@@ -0,0 +1,106 @@
+package ns1
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceRecord() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ttl": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"meta": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+			},
+			"link": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
Looks like there are still several attributes in this schema that still contain `Optional: true` and are missing `Computed: true`.

 _All_ attributes in a data source that are not configurable (in this case everything other than `zone` `domain` and `type`) should have `Computed: true` set and not `Optional: true` or `Required: true`.  Apologies if I was unclear about that in my earlier comments.

Reply:
No, that's fine - more than happy to make those changes 🙂

Reply:
Changes made & my little test above still works!
If there's anymore changes, please let me know - more than happy to learn! Will come in handy as I make more changes to this provider 😃